pkg/config/viperconfig/viper.go

```diff
@@ -102,17 +101,18 @@ func (c *safeConfig) Set(key string, newValue interface{}, source model.Source)
 	latestValue := c.Viper.Get(key)
 	if !reflect.DeepEqual(oldValue, latestValue) {
 		log.Debugf("Updating setting '%s' for source '%s' with new value. notifying %d listeners", key, source, len(c.notificationReceivers))
-		// if the value has not changed, do not duplicate the slice so that no callback is called
-		receivers = slices.Clone(c.notificationReceivers)
 	} else {
+		// if the value has not changed, do not notify the receivers
 		log.Debugf("Updating setting '%s' for source '%s' with the same value, skipping notification", key, source)
+		return
 	}
-	c.Unlock()
+	// Increment the sequence ID for the key
```

Comment:
Comment is slightly stale now that there's no per-key sequence ID... could probably just drop it since we don't have an equivalent comment for the NTM version.